### PR TITLE
Feature/azure monitoring optional

### PR DESCRIPTION
**Pull request by arbulu89**
Add the option to disable the monitoring stack in Azure.

Fix: https://github.com/SUSE/ha-sap-terraform-deployments/issues/182

### CODE REVIEW

**Comment by MalloZup** [comment]
Regarding doc, I think we need absolutely to highlight this magic vars which create new hosts. They are different from the others vars so we need a special section perhaps in main Readme

**Comment by arbulu89** [comment]
@MalloZup It looks like a basic `enable/disable` boolean variable for me hehe. Anyway, I will add a small text in the `monitoring` doc mentioning that.

**Comment by MalloZup** [comment]
@arbulu89  I'm not sure if we should add it only on monitoring. Perhaps we should add a section on readme but I'm ok to add in monitoring and think later.

File: azure/image.tf [hunk 1]
```diff
@@ -41,12 +41,12 @@ variable "iscsi_public_offer" {
 
 variable "iscsi_public_sku" {
   type    = string
-  default = "12-sp4"
+  default = "15"
```
[the remark this answers is not in the document]
Comment (MalloZup) [reply azure/image.tf]: @juadk  can we open a bug for this if we don't have it? A github issue would be enough.

SLE12sp4 should work in middle term

**Comment by arbulu89** [comment]
@juadk @MalloZup I have added some more context in the documentation. I wouldn't add this information to the main README. Now, the monitoring stack is `False` by default, so if anyone wants to add it, I guess they would need to check the documentation.

File: azure/image.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (arbulu89) [reply azure/image.tf]: This is something related to the formula and to python dependencies as far as I remember. In any case, we should open the ticket in: https://github.com/saltstack-formulas/iscsi-formula

[the remark this answers is not in the document]
Comment (juadk) [reply azure/image.tf]: > This is something related to the formula and to python dependencies as far as I remember. In any case, we should open the ticket in: https://github.com/saltstack-formulas/iscsi-formula

Exactly and I have to test [this](https://github.com/saltstack-formulas/iscsi-formula/pull/23) as well.

[the remark this answers is not in the document]
Comment (MalloZup) [reply azure/image.tf]: @arbulu89  we could open a ticket upstream and reference it here. 

see as @juadk  did already with:

- https://github.com/terraform-providers/terraform-provider-azurerm/issues/4361
 which point to https://github.com/SUSE/ha-sap-terraform-deployments/issues/188

So we gain upstream visibility :grapes:

**Comment by MalloZup** [comment]
@arbulu89 ok fine for moment for me. thx!

**Comment by ldevulder** [comment]
I just do regression tests with both value of `background` variable for AWS/GCP: OK!

**Comment by ldevulder** [comment]
I saw what can be a small issue: `monitoring_srv_ip` needs to be set even if I set `monitoring_enabled`. is it expected? Anyway setting it to `""` did the job.

I'm still testsing on Azure, I have sporadic issue (as usual!) that delay the verification...

File: azure/image.tf [hunk 2]
```diff
@@ -41,12 +41,12 @@ variable "iscsi_public_offer" {
...
 }
 
 variable "iscsi_public_version" {
   type    = string
-  default = "2019.03.06"
+  default = "2019.07.17"
```
Comment (diegoakechi) [inline azure/image.tf]: @arbulu89 Azure supports `latest` as a valid option. I think this is a better default to keep the deployment always with the most up-to-date version of the image. Not critical for this PR though.

**Comment by diegoakechi** [comment]
> I saw what can be a small issue: `monitoring_srv_ip` needs to be set even if I set `monitoring_enabled`. is it expected? Anyway setting it to `""` did the job.

@arbulu89 I think we can set `default = ""` for this variable to avoid this inconvenience, since disabled is the default for the monitoring module.

File: azure/image.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (arbulu89) [reply azure/image.tf]: @MalloZup This issue doesn't have any relation with iscsi-formula one, right?
I'm a bit confused.

**Comment by ldevulder** [comment]
I still have sporadic issue with Azure... @juadk will try to test this. So just a small delay for the validation, sorry!

File: azure/image.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (MalloZup) [reply azure/image.tf]: yep just an example :)

**Comment by juadk** [comment]
> I still have sporadic issue with Azure... @juadk will try to test this. So just a small delay for the validation, sorry!

It works well for me in Azure.

**Pull request merged**